mysql-test/suite/parts/t/mdev-21027.test

```diff
@@ -0,0 +1,23 @@
+#
+# MDEV-21027 Assertion `part_share->auto_inc_initialized || !can_use_for_auto_inc_init()' failed in ha_partition::set_auto_increment_if_higher
+#
+
+--source include/have_partition.inc
+
+CREATE TABLE t1 (a INT AUTO_INCREMENT PRIMARY KEY) PARTITION BY HASH (a) PARTITIONS 3;
+--error ER_ROW_DOES_NOT_MATCH_GIVEN_PARTITION_SET
+REPLACE INTO t1 PARTITION (p0) VALUES (3),(8);
```

Comment:
Ask for the reviewer: This test code came from the issue ticket. But actually, before this commit, this query also fails for the same reason:
```
REPLACE INTO t1 PARTITION (p0) VALUES (3);
```
And after this commit, this query is executed successfully. On the other hand, the current query is expected to fail (so I added an --error flag).
I think that both queries are suitable for checking the bug.

I am wondering which test code is better for us.

Reply:
If both are OK for testing, I believe that the simpler one, here `REPLACE INTO t1 PARTITION (p0) VALUES (3)`, is better. Thank you for trying to minimize the test case. This is a great practice!